library/include/G4CMPSolidUtils.hh

```diff
@@ -0,0 +1,172 @@
+/***********************************************************************\
+ * This software is licensed under the terms of the GNU General Public *
+ * License version 3 or later. See G4CMP/LICENSE for the full license. *
+\***********************************************************************/
+
+/// \file library/include/G4CMPSolidUtils.hh
+/// \brief This class contains custom utilities for G4VSolids that are not
+/// directly included in the G4VSolid class. Some example of utilites are:
+/// 1) Adjusting a position to the nearest surface point without the surface normal.
+/// 2) Adjusting a particle to the nearest edge along the surface.
+///
+/// The user will need to create this class directly with a G4VSolid and/or
+/// touchable, or can initialize these objects later on.
+///
+/// Note: All coordinates/vectors passed into functions must be in the solid's local coordinate system.
```

Comment:
You might append to this, "If a global-to-local transform is supplied, the input coordinates and vectors, and any output results, will be in the coordinate system of that transform."